packages/grafana-flamegraph/src/FlameGraph/FlameGraph.tsx

```diff
@@ -66,30 +66,42 @@ const FlameGraph = ({
 }: Props) => {
   const styles = getStyles();
 
-  const [collapsedMap, setCollapsedMap] = useState<CollapsedMap>(data ? data.getCollapsedMap() : new Map());
+  const [collapsedMap, setCollapsedMap] = useState<CollapsedMap>(new Map());
+  const [levels, setLevels] = useState<LevelItem[][]>();
+  const [levelsCallers, setLevelsCallers] = useState<LevelItem[][]>();
+  const [totalProfileTicks, setTotalProfileTicks] = useState<number>(0);
+  const [totalProfileTicksRight, setTotalProfileTicksRight] = useState<number>();
+  const [totalViewTicks, setTotalViewTicks] = useState<number>(0);
+
   useEffect(() => {
     if (data) {
       setCollapsedMap(data.getCollapsedMap());
+
+      let levels = data.getLevels();
```

Comment:
Instead of useMap for the collapseMap and separate useEffect for the levels computation this is done in single effect which synchronizes those 2 values and prevent flickering on double rerenders.